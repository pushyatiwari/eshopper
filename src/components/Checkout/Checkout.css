.email-validate-true{
  visibility: hidden;
  height: 0.1rem;
}
.email-validate-false{
    color: red;
    height: 0.1rem;
}
.form-container{
    display: flex;
    justify-content:center;
    flex-direction: column;
    align-content: center;
    margin: 10rem;
    border-style: solid;
    border-width: 0.1rem;
    padding:10%;
    width: 12rem;
    border-radius: 10px;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);

}

.form-element{
    display: flex;
    justify-content: space-between;
}

