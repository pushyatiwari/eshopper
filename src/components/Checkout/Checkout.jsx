/* eslint-disable react/prop-types */
/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
// import * as EmailValidator from 'email-validator';
import './Checkout.css';

const Checkout = ({ checkout }) => {
  const [formname, setName] = useState('');
  const [formemail, setEmail] = useState('');
  const [formaddress, setAddress] = useState('');
  const [formphone, setPhone] = useState('');
  const [isOrdered, setisOrdered] = useState(false);
  const handleChange = (event) => {
    const { name, value } = event.target;
    // this.setState({ [name]: value });
    if (name === 'formname') setName(value);
    if (name === 'formemail') setEmail(value);
    if (name === 'formaddress') setAddress(value);
    if (name === 'formphone') setPhone(value);
  };
  const handleSubmit = (evt) => {
    evt.preventDefault();
    checkout();
    setisOrdered(true);
    console.log('order placed');
    // console.log(formname);
    // console.log(formemail);
    // console.log(formaddress);
    // console.log(formphone);
  };

  return (
    <>
      <div>
        <form className="form-container" onSubmit={handleSubmit}>
          <div className="form-element">
            <p>NAME:</p>
            {' '}
            <input data-testid="input-name" type="text" name="formname" aria-describedby="name-format" required aria-required="true" onChange={handleChange} value={formname} pattern="[A-Za-z-0-9]+\s[A-Za-z-'0-9]+" />
            <br />
          </div>
          <div className="form-element">
            <p>EMAIL:</p>
            <input data-testid="input-email" type="email" id="formemail" name="formemail" required onChange={handleChange} value={formemail} pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$" />
            <br />
          </div>
          <div className="form-element">
            <p>ADDRESS:</p>
            <input data-testid="input-address" type="text" name="formaddress" onChange={handleChange} value={formaddress} />
            <br />
          </div>
          <div className="form-element">
            <p>PHONE:</p>
            <input
              data-testid="input-phone"
              type="text"
              name="formphone"
              onChange={handleChange}
              value={formphone}
              pattern="^\d{10}$"
              required="required"
            />
          </div>
          <br />
          <button type="submit">Checkout</button>
        </form>
      </div>
      <h1 className={isOrdered ? 'order-placed' : 'order-not-placed'}>Your Order is placed successfully!</h1>
    </>
  );
};

export default Checkout;
