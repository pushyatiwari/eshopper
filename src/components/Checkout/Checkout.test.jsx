import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import Checkout from './Checkout';

describe(Checkout.name, () => {
  test('should set the name', () => {
    render(<Checkout />);
    const nameInput = screen.getByTestId('input-name');
    fireEvent.change(nameInput, { target: { value: 'pushya' } });
    expect(nameInput.value).toBe('pushya');
  });
  //   test('should set the email', () => {
  //     render(<Checkout />);
  //     const nameInput = screen.getByTestId('input-name');
  //     fireEvent.change(nameInput, { target: { value: 'pushya' } });
  //     expect(nameInput.value).toBe('pushya');
  //   });
  test('should set the email', () => {
    render(<Checkout />);
    const nameInput = screen.getByTestId('input-name');
    fireEvent.change(nameInput, { target: { value: 'pushyatiwari@gmail.com' } });
    expect(nameInput.value).toBe('pushyatiwari@gmail.com');
  });
  test('should set the phone', () => {
    render(<Checkout />);
    const nameInput = screen.getByTestId('input-email');
    fireEvent.change(nameInput, { target: { value: 'pushyatiwari@gmail.com' } });
    expect(nameInput.value).toBe('pushyatiwari@gmail.com');
  });
  test('should set the number', () => {
    render(<Checkout />);
    const nameInput = screen.getByTestId('input-phone');
    fireEvent.change(nameInput, { target: { value: '9846473809' } });
    expect(nameInput.value).toBe('9846473809');
  });
  test('should be called with', () => {
    const checkoutContainer = render(<Checkout />);
    expect(checkoutContainer).toMatchSnapshot();
  });
});
