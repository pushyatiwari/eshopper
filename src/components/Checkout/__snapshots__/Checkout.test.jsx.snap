// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout should be called with 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form
          class="form-container"
        >
          <div
            class="form-element"
          >
            <p>
              NAME:
            </p>
             
            <input
              aria-describedby="name-format"
              aria-required="true"
              data-testid="input-name"
              name="formname"
              pattern="[A-Za-z-0-9]+\\\\s[A-Za-z-'0-9]+"
              required=""
              type="text"
              value=""
            />
            <br />
          </div>
          <div
            class="form-element"
          >
            <p>
              EMAIL:
            </p>
            <input
              data-testid="input-email"
              id="formemail"
              name="formemail"
              pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
              required=""
              type="email"
              value=""
            />
            <br />
          </div>
          <div
            class="form-element"
          >
            <p>
              ADDRESS:
            </p>
            <input
              data-testid="input-address"
              name="formaddress"
              type="text"
              value=""
            />
            <br />
          </div>
          <div
            class="form-element"
          >
            <p>
              PHONE:
            </p>
            <input
              data-testid="input-phone"
              name="formphone"
              pattern="^\\\\d{10}$"
              required=""
              type="text"
              value=""
            />
          </div>
          <br />
          <button
            type="submit"
          >
            Checkout
          </button>
        </form>
      </div>
      <h1
        class="order-not-placed"
      >
        Your Order is placed successfully!
      </h1>
    </div>
  </body>,
  "container": <div>
    <div>
      <form
        class="form-container"
      >
        <div
          class="form-element"
        >
          <p>
            NAME:
          </p>
           
          <input
            aria-describedby="name-format"
            aria-required="true"
            data-testid="input-name"
            name="formname"
            pattern="[A-Za-z-0-9]+\\\\s[A-Za-z-'0-9]+"
            required=""
            type="text"
            value=""
          />
          <br />
        </div>
        <div
          class="form-element"
        >
          <p>
            EMAIL:
          </p>
          <input
            data-testid="input-email"
            id="formemail"
            name="formemail"
            pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\\\\.[a-z]{2,4}$"
            required=""
            type="email"
            value=""
          />
          <br />
        </div>
        <div
          class="form-element"
        >
          <p>
            ADDRESS:
          </p>
          <input
            data-testid="input-address"
            name="formaddress"
            type="text"
            value=""
          />
          <br />
        </div>
        <div
          class="form-element"
        >
          <p>
            PHONE:
          </p>
          <input
            data-testid="input-phone"
            name="formphone"
            pattern="^\\\\d{10}$"
            required=""
            type="text"
            value=""
          />
        </div>
        <br />
        <button
          type="submit"
        >
          Checkout
        </button>
      </form>
    </div>
    <h1
      class="order-not-placed"
    >
      Your Order is placed successfully!
    </h1>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
